src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -0,0 +1,376 @@
+            logger.error(
+                f"Template for resource group {resource_group} "
+                f"has failed validation. The message was: "
+                f"{validation_res.error.message}. See logs for additional details."
+            )
+            logger.debug(
+                f"Template for resource group {resource_group} "
+                f"failed validation. Full error details: {validation_res.error}."
+            )
+            raise RuntimeError("Azure template validation failed.")
+
+        # Validation succeeded so proceed with deployment
+        logger.debug(f"Successfully validated resources for {resource_group}")
+
+        poller = self.api_clients.resource_client.deployments.begin_create_or_update(
+            resource_group_name=resource_group,
+            deployment_name=deployment_name,
+            parameters={
+                "properties": {
+                    "mode": "Incremental",
+                    "template": template,
+                    "parameters": parameters,
+                }
+            },
+        )
+        logger.debug(poller)
+
+        # Wait for the deployment to complete and get the outputs
+        deployment: DeploymentExtended = poller.result()
+        logger.debug("Finished deploying")
+
+        if deployment.properties is not None:
+            depl_props = deployment.properties
+        else:
+            raise RuntimeError("The deployment has no properties.\nAborting")
+        logger.debug(f"Deployed: {deployment.name} {deployment.id} {depl_props}")
+
+        if depl_props.provisioning_state != "Succeeded":
+            logger.debug(f"Failed to provision: {depl_props}")
+            raise RuntimeError(
+                f"Deploy of template to resource group"
+                f" {resource_group} proceeded but the provisioning"
+                f" state returned is {depl_props.provisioning_state}. "
+                f"\nAborting"
+            )
+        logger.debug(
+            f"Provisioning state of {resource_group}"
+            f": {depl_props.provisioning_state}"
+        )
+
+        return depl_props.outputs
+
+    def convert_bicep_to_arm(self, bicep_template: str) -> Any:
+        """
+        Convert a bicep template into an ARM template.
+
+        :param bicep_template: The path to the bicep template to be converted
+
+        :raise RuntimeError if az CLI is not installed.
+        :return: Output dictionary from the bicep template.
+        """
+        if not shutil.which("az"):
+            logger.error(
+                "The Azure CLI is not installed - follow "
+                "https://github.com/Azure/bicep/blob/main/docs/installing.md#linux"
+            )
+            raise RuntimeError(
+                "The Azure CLI is not installed - cannot render ARM templates."
+            )
+        logger.debug(f"Converting {bicep_template} to ARM template")
+
+        arm_template_name = bicep_template.replace(".bicep", ".json")
```

Comment:
Though the risk is low given the provenance of this file, it's probably safest to use tempfile or such to ensure this is being done on a temporary file that shouldn't have any risk of already existing.